### PR TITLE
Initial commit with complete homework requirements

### PR DESCRIPTION
**Pull request by bengu-thon-mai-mochi**
- Use of States
- Use of Lifecycle Methods
- Setting counter for browser

- Buttons and their functionalities : 
  *Checkbox button : Change style with in-line ternary operators.
  *Delete button  
  *Add button

### CODE REVIEW

File: react/week2/todo-list/src/Counter.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import React, {Component} from 'react';
+
+class Counter extends Component {
+  constructor(){
+    super()
+    this.state = {counter: 0}
+  };
```
Comment (braedongough) [inline react/week2/todo-list/src/Counter.js:7]: In the version of react we're using, you can omit the constructor and super, it's done behind the scenes, so you could re-write this like: 
```suggestion
    state = { counter: 0 };
```
You can read more about it in the [docs](https://reactjs.org/docs/react-component.html#constructor), however I will admit, they are a bit confusing to understand.

File: react/week2/todo-list/src/Counter.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  componentDidMount = () => {
+    this.timer = setInterval(
+      () => this.incrementCounter(),
+    1000
+    );
+  };
+
+  componentWillUnmount() {
+    clearInterval(this.timer);
+  }
```
Comment (braedongough) [inline react/week2/todo-list/src/Counter.js:18]: Nice job remember to clean up upon unmount! This is easy to forget :)

File: react/week2/todo-list/src/Counter.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  constructor(){
+    super()
+    this.state = {counter: 0}
+  };
+  
+  componentDidMount = () => {
+    this.timer = setInterval(
+      () => this.incrementCounter(),
+    1000
+    );
```
Comment (braedongough) [inline react/week2/todo-list/src/Counter.js:13]: If you think about it, `incrementCounter` is already an arrow function, so you could pass this in simple as a reference to the function like: 
```suggestion
    this.timer = setInterval(this.incrementCounter, 1000);
```

File: react/week2/todo-list/src/TodoItem.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import './TodoItem.css';
+
+function TodoItem(props){
+    const {description, id, deleteTodo, todoItemClicked} = props;
+    const { isChecked } = props;
+
+    const todoItemChecked = event => {
```
Comment (braedongough) [inline react/week2/todo-list/src/TodoItem.js:8]: I think it's more clear to write your changeHandlers as something that will happen, as opposed to something that has happened: 
```suggestion
    const checkTodoItem = event => {
```

File: react/week2/todo-list/src/TodoList.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    };
+
+    addTodo = () => {
+        const { todos } = this.state;
+        const newTodo = {
+            description: "Random text",
+            id: new Date(),
+            isChecked: false
+        }
+        this.setState({ todos: [...this.state.todos, newTodo]});
```
Comment (braedongough) [inline react/week2/todo-list/src/TodoList.js:28]: Nice use of the spread operator!

File: react/week2/todo-list/src/Counter.js (hunk 1 above)
Comment (bengu-thon-mai-mochi) [reply react/week2/todo-list/src/Counter.js:7]: it's week2 homework. 
implemented :)

**Pull request merged**